src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -82,6 +82,8 @@ protected void TypeChecks(TypeSymbol type, DiagnosticBag diagnostics)
                 diagnostics.Add(ErrorCode.ERR_VolatileStruct, this.ErrorLocation, this, type);
             }
 
+            type.CheckRecordLangVersion(diagnostics, TypeSyntax, TypeSyntax?.Location);
```

Comment:
>TypeSyntax?.Location [](start = 65, length = 20)

Why conditional access? It doesn't look like the helper is expecting null as location.